src/main/java/net/snowflake/client/core/SFTrustManager.java

```diff
@@ -559,12 +559,21 @@ private static String byteToHexString(byte[] bytes) {
    * @return HttpClient
    */
   private static CloseableHttpClient getHttpClient(int timeout) {
-    RequestConfig config =
+    RequestConfig.Builder configBuilder =
         RequestConfig.custom()
             .setConnectTimeout(timeout)
             .setConnectionRequestTimeout(timeout)
-            .setSocketTimeout(timeout)
-            .build();
+            .setSocketTimeout(timeout);
+    HttpHost proxy = null;
+    if (proxySettingsKey.usesProxy()) {
+      proxy =
+          new HttpHost(
+              proxySettingsKey.getProxyHost(),
+              proxySettingsKey.getProxyPort(),
+              proxySettingsKey.getProxyProtocol().toString());
+      configBuilder.setProxy(proxy);
```

Comment:
Will this force the connection to use TLS if `getProxyProtocol` returns HTTPS?

Reply:
Removed trust manager code